src/back/spv/block.rs

```diff
@@ -1334,15 +1334,35 @@ impl<'w> BlockContext<'w> {
                 block.body.push(instruction);
                 id
             }
-            crate::Expression::Derivative { axis, expr } => {
-                use crate::DerivativeAxis as Da;
-
+            crate::Expression::Derivative { axis, ctrl, expr } => {
+                use crate::{DerivativeAxis as Axis, DerivativeControl as Ctrl};
+                match ctrl {
+                    Ctrl::Coarse | Ctrl::Fine => {
+                        self.writer.require_any(
+                            "DerivativeControl",
+                            &[spirv::Capability::DerivativeControl],
+                        )?;
+                    }
+                    Ctrl::None => {}
+                }
                 let id = self.gen_id();
                 let expr_id = self.cached[expr];
                 let op = match axis {
-                    Da::X => spirv::Op::DPdx,
-                    Da::Y => spirv::Op::DPdy,
-                    Da::Width => spirv::Op::Fwidth,
+                    Axis::X => match ctrl {
+                        Ctrl::Coarse => spirv::Op::DPdxCoarse,
+                        Ctrl::Fine => spirv::Op::DPdxFine,
+                        Ctrl::None => spirv::Op::DPdx,
+                    },
+                    Axis::Y => match ctrl {
+                        Ctrl::Coarse => spirv::Op::DPdyCoarse,
+                        Ctrl::Fine => spirv::Op::DPdyFine,
+                        Ctrl::None => spirv::Op::DPdy,
+                    },
+                    Axis::Width => match ctrl {
+                        Ctrl::Coarse => spirv::Op::FwidthCoarse,
+                        Ctrl::Fine => spirv::Op::FwidthFine,
+                        Ctrl::None => spirv::Op::Fwidth,
+                    },
```

Comment:
It would be nicer if we only had one match here, testing for both the axis and ctrl at the same time (i.e. `match (axis, ctrl)`). This applies to a few other places in the other backends as well.

Reply:
Appreciate the feedback, will get to these shortly